consensus/istanbul/core/preprepare.go

```diff
@@ -75,20 +89,30 @@ func (c *core) handlePreprepare(msg *message, src istanbul.Validator) error {
 	}
 
 	// Check if the message comes from current proposer
-	if !c.valSet.IsProposer(src.Address()) {
+	if !c.valSet.IsProposer(msg.Address) {
 		logger.Warn("Ignore preprepare messages from non-proposer")
 		return errNotFromProposer
 	}
 
+	if preprepare.View.Round.Cmp(common.Big0) > 0 {
+		// If we have a PREPARED certificate after handling the ROUND CHANGE, the proposal must match.
+		// TODO(asa): Does it make a difference if this PREPARED certificate came from the ROUND CHANGE certificate vs
```

Comment:
Wdyt @trianglesphere?

Reply:
I don't think that it makes a difference.

When a node handles round change messages, it resets its `prepared certificate`, thus the proposals should match. I think we may need to double check to see that we make sure we start the new round at the correct time.